### PR TITLE
fix(CBH): upgrade CBH resource API version from V1 to V2

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

- The API version currently used by CBH resources is V1, and the V1 version of the API will be offline soon. So we need to upgrade the version to V2.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- Please view all code in the changed files.
- For specific business logic change basis, please refer to the design document.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbh' TESTARGS='-run TestAccCBHInstance_basic'
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1088.25s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1088.291s
```

### CODE REVIEW

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cbh' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1073.57s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1073.610s
```

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 1]
```diff
@@ -175,448 +170,288 @@ func ResourceCBHInstance() *schema.Resource {
...
 	}
-	getCbhOrderProductIdOpt.JSONBody = utils.RemoveNil(buildGetCbhFlavorsBodyParams(d,
-		getCbhOrderProductIdClient.ProjectID, region))
-	getCbhOrderProductIdResp, err := getCbhOrderProductIdClient.Request("POST",
-		getCbhOrderProductIdPath, &getCbhOrderProductIdOpt)
+	getCbhInstancesResp, err := client.Request("GET", getCbhInstancesPath, &getCbhInstancesOpt)
 
 	if err != nil {
-		return "", fmt.Errorf("error getting CBH order product id: %s", err)
+		return nil, err
```
Comment (jinyangyang222) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:423]: You can add an error message here to distinguish it from the error in line 428

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 2]
```diff
@@ -625,193 +460,147 @@ func resourceCBHInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta
...
+	instances, err := getCBHInstanceList(client)
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	if len(instances) == 0 {
+
+	expression := fmt.Sprintf("[?server_id == '%s']|[0]", d.Id())
+	instance := utils.PathSearch(expression, instances, nil)
+	if instance == nil {
 		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
```
Comment (jinyangyang222) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:582]:
```suggestion
		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "CBH instance")
```

File: docs/resources/cbh_instance.md [hunk 3]
```diff
@@ -41,7 +41,8 @@ The following arguments are supported:
 
   Changing this parameter will create a new resource.
 
-* `name` - (Required, String, ForceNew) Specifies the name of the CBH instance.
+* `name` - (Required, String, ForceNew) Specifies the name of the CBH instance. The field can contain `1` to `64` characters.
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply docs/resources/cbh_instance.md]: OK

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 4]
```diff
@@ -175,448 +170,288 @@ func ResourceCBHInstance() *schema.Resource {
...
+		"password":          d.Get("password"),
+		"region":            region,
+		"availability_zone": d.Get("availability_zone"),
+		"charging_mode":     buildCreateChargingModeParam(d),
+		"period_type":       buildCreatePeriodTypeParam(d),
+		"period_num":        utils.ValueIngoreEmpty(d.Get("period")),
+		"is_auto_renew":     buildCreateIsAutoRenewParam(d),
+		"is_auto_pay":       1,
+		"network":           buildCreateNetworkBodyParam(d, publicIp),
+		"ipv6_enable":       d.Get("ipv6_enable"),
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]: OK

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 5]
```diff
@@ -625,193 +464,147 @@ func resourceCBHInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta
...
-	}
-	unbindEipRespBody, err := utils.FlattenResponse(unbindEipResp)
+	_, err := client.Request("POST", bindPath, &unbindEipOpt)
 	if err != nil {
-		return err
-	}
-	unbindEipRespBodyBytes, _ := json.Marshal(unbindEipRespBody)
-	if !strings.Contains(string(unbindEipRespBodyBytes), "success") {
-		return fmt.Errorf("fail unbind eip from CBH instance: %s", string(unbindEipRespBodyBytes))
+		return fmt.Errorf("error bind EIP (%s) from CBH instance (%s): %s", publicIpId, d.Id(), err)
```
Comment (houpeng80) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]:
```suggestion
		return fmt.Errorf("error bind EIP (%s) to CBH instance (%s): %s", publicIpId, d.Id(), err)
```

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 6]
```diff
@@ -175,448 +170,292 @@ func ResourceCBHInstance() *schema.Resource {
...
+			status := utils.PathSearch("status_info.status", instance, "").(string)
+			if status == "ACTIVE" {
+				return instance, "COMPLETED", nil
+			}
+
+			if status == "" {
+				return instance, "ERROR", fmt.Errorf("status is not found in list API response")
+			}
+
+			unexpectedStatus := []string{"SHUTOFF", "DELETING", "DELETED", "ERROR", "FROZEN"}
```
Comment (houpeng80) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]: move to line 371?

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cbh' TESTARGS='-run TestAccCBHInstance_basic'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1064.03s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1064.069s
```

**Pull request merged**